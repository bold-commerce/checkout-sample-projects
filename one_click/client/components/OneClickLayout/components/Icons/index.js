export { default as TimesCircle } from './TimesCirlce'
export { default as ArrowRight } from './ArrowRight'
export { default as CaretRight } from './CaretRight'