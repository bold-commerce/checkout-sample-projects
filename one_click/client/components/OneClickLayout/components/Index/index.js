export { default as IndexPage } from "./IndexPage";
export { default as IndexGuest } from "./IndexGuest";