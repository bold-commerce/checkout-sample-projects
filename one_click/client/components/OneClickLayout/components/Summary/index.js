export { default as Summary } from './Summary';
export { default as SummaryCondensed } from './SummaryCondensed';