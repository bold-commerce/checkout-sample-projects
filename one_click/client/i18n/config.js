import i18n from 'i18next';
import enTranslation from './en/translation.json';
import { initReactI18next } from 'react-i18next';

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: {
        translation: enTranslation
      }
    },
    lng: 'en',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false, 
    }
  });

export default i18n;