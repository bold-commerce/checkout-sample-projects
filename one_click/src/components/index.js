export * from './Address';
export * from './BackButton';
export * from './Billing';
export * from './BillingAddress';
export * from './Card';
export * from './CheckoutButton';
export * from './CheckoutForm';
export * from './Confirmation';
export * from './CustomerInfo';
export * from './Discount';
export * from './EmptyState';
export * from './Header';
export * from './Icons';
export * from './Index';
export * from './InventoryItem';
export * from './LineItems';
export * from './LoadingState';
export * from './Payment';
export * from '../pages/Processing';
export * from './Product';
export * from './RadioItem';
export * from './Shipping';
export * from './ShippingLines';
export * from './Summary';
