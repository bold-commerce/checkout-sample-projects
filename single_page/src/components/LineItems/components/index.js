export { default as LineItem } from './LineItem';
export { default as Product } from './Product';
