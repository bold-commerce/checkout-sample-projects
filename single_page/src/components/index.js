export * from './Address';
export * from './BillingAddress';
export * from './CheckoutButton';
export * from './CheckoutSection';
export * from './Customer';
export * from './DiscountForm';
export * from './EmptyState';
export * from './Header';
export * from './LineItems';
export * from './LoadingState';
export * from './OrderErrors';
export * from './OrderSummary';
export * from './PaymentMethod';
export * from './ShippingAddress';
export * from './ShippingLines';
