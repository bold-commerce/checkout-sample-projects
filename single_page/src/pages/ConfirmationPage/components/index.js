export * from './ConfirmationList';
export * from './ConfirmationListItem';
