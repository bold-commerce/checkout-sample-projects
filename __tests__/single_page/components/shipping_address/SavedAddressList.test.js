import React from "react";
import { render } from "@testing-library/react";
import { SavedAddressList } from '../../../../single_page/src/components/ShippingAddress';
import { exampleSavedAddresses as MOCKaddresses } from '../../../utils/addressHelpers';
import '../../../../single_page/src/i18n/config';

describe('ShippingAddress', () => {
  test('renders ShippingAddress component', () => {
    const { asFragment } = render(
      <SavedAddressList 
        savedAddresses={MOCKaddresses}
      /> );

    expect(asFragment()).toMatchSnapshot();
  });

  test('renders ShippingAddress component with selected address', () => {
    const { asFragment } = render(
      <SavedAddressList 
        savedAddresses={MOCKaddresses}
        selectedAddress={2}
      /> );

    expect(asFragment()).toMatchSnapshot();
  });

  test('renders ShippingAddress component disabled', () => {
    const { asFragment } = render(
      <SavedAddressList 
        savedAddresses={MOCKaddresses}
        disabled
      /> );

    expect(asFragment()).toMatchSnapshot();
  });
});
