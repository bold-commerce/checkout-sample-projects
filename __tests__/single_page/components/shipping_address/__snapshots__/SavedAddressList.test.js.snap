// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingAddress renders ShippingAddress component 1`] = `
<DocumentFragment>
  <div
    class="RadioButton RadioButton__AddressInfoContainer"
  >
    <div
      class="stx-field stx-field--with-radio-field RadioField"
    >
      <div
        class="stx-radio-field"
      >
        <span
          class="stx-radio stx-radio-field__radio"
        >
          <input
            aria-checked="false"
            class="stx-radio__input"
            id="stxField1"
            name="shipping-address"
            role="radio"
            type="radio"
            value=""
          />
          <span
            class="stx-radio__box"
          />
        </span>
        <label
          class="stx-radio-field__label"
          for="stxField1"
        >
          <div
            class="CheckoutStep__FieldSetInfoContainer"
          >
            <span
              class="CheckoutStep__FieldSetInfo"
            >
              John Doe
            </span>
            <span
              class="CheckoutStep__FieldSetInfo CheckoutStep__FieldSetInfo--Secondary"
            >
              
        123 Fake Street,
        Unit 12,
        Winnipeg,
        Manitoba,
        R3A1B3,
        Canada,
        204-123-4567
        
            </span>
          </div>
        </label>
      </div>
    </div>
  </div>
  <div
    class="RadioButton RadioButton__AddressInfoContainer"
  >
    <div
      class="stx-field stx-field--with-radio-field RadioField"
    >
      <div
        class="stx-radio-field"
      >
        <span
          class="stx-radio stx-radio-field__radio"
        >
          <input
            aria-checked="false"
            class="stx-radio__input"
            id="stxField2"
            name="shipping-address"
            role="radio"
            type="radio"
            value=""
          />
          <span
            class="stx-radio__box"
          />
        </span>
        <label
          class="stx-radio-field__label"
          for="stxField2"
        >
          <div
            class="CheckoutStep__FieldSetInfoContainer"
          >
            <span
              class="CheckoutStep__FieldSetInfo"
            >
              John Doe
            </span>
            <span
              class="CheckoutStep__FieldSetInfo CheckoutStep__FieldSetInfo--Secondary"
            >
              
        456 Test Drive,
        
        Winnipeg,
        Manitoba,
        R3A1B3,
        Canada,
        204-123-4567
        
            </span>
          </div>
        </label>
      </div>
    </div>
  </div>
  <div
    class="RadioButton RadioButton__AddressInfoContainer"
  >
    <div
      class="stx-field stx-field--with-radio-field RadioField"
    >
      <div
        class="stx-radio-field"
      >
        <span
          class="stx-radio stx-radio-field__radio"
        >
          <input
            aria-checked="false"
            class="stx-radio__input"
            id="stxField3"
            name="shipping-address"
            role="radio"
            type="radio"
            value=""
          />
          <span
            class="stx-radio__box"
          />
        </span>
        <label
          class="stx-radio-field__label"
          for="stxField3"
        >
          <div
            class="CheckoutStep__FieldSetInfoContainer"
          >
            <span
              class="CheckoutStep__FieldSetInfo"
            >
              John Doe
            </span>
            <span
              class="CheckoutStep__FieldSetInfo CheckoutStep__FieldSetInfo--Secondary"
            >
              
        789 Fake Crescent,
        
        Winnipeg,
        Manitoba,
        R3A1B3,
        Canada,
        204-123-4567
        
            </span>
          </div>
        </label>
      </div>
    </div>
  </div>
  <div
    class="RadioButton RadioButton__NewAddressContainer"
  >
    <div
      class="stx-field stx-field--with-radio-field RadioField"
    >
      <div
        class="stx-radio-field"
      >
        <span
          class="stx-radio stx-radio--is-checked stx-radio-field__radio"
        >
          <input
            aria-checked="true"
            checked=""
            class="stx-radio__input"
            id="stxField4"
            name="shipping-address"
            role="radio"
            type="radio"
            value=""
          />
          <span
            class="stx-radio__box"
          />
        </span>
        <label
          class="stx-radio-field__label"
          for="stxField4"
        >
          Add a new address
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ShippingAddress renders ShippingAddress component disabled 1`] = `
<DocumentFragment>
  <div
    class="RadioButton RadioButton__AddressInfoContainer"
  >
    <div
      class="stx-field stx-field--is-disabled stx-field--with-radio-field RadioField"
    >
      <div
        class="stx-radio-field"
      >
        <span
          class="stx-radio stx-radio--is-disabled stx-radio-field__radio"
        >
          <input
            aria-checked="false"
            aria-disabled="true"
            class="stx-radio__input"
            disabled=""
            id="stxField9"
            name="shipping-address"
            role="radio"
            type="radio"
            value=""
          />
          <span
            class="stx-radio__box"
          />
        </span>
        <label
          class="stx-radio-field__label"
          for="stxField9"
        >
          <div
            class="CheckoutStep__FieldSetInfoContainer"
          >
            <span
              class="CheckoutStep__FieldSetInfo"
            >
              John Doe
            </span>
            <span
              class="CheckoutStep__FieldSetInfo CheckoutStep__FieldSetInfo--Secondary"
            >
              
        123 Fake Street,
        Unit 12,
        Winnipeg,
        Manitoba,
        R3A1B3,
        Canada,
        204-123-4567
        
            </span>
          </div>
        </label>
      </div>
    </div>
  </div>
  <div
    class="RadioButton RadioButton__AddressInfoContainer"
  >
    <div
      class="stx-field stx-field--is-disabled stx-field--with-radio-field RadioField"
    >
      <div
        class="stx-radio-field"
      >
        <span
          class="stx-radio stx-radio--is-disabled stx-radio-field__radio"
        >
          <input
            aria-checked="false"
            aria-disabled="true"
            class="stx-radio__input"
            disabled=""
            id="stxField10"
            name="shipping-address"
            role="radio"
            type="radio"
            value=""
          />
          <span
            class="stx-radio__box"
          />
        </span>
        <label
          class="stx-radio-field__label"
          for="stxField10"
        >
          <div
            class="CheckoutStep__FieldSetInfoContainer"
          >
            <span
              class="CheckoutStep__FieldSetInfo"
            >
              John Doe
            </span>
            <span
              class="CheckoutStep__FieldSetInfo CheckoutStep__FieldSetInfo--Secondary"
            >
              
        456 Test Drive,
        
        Winnipeg,
        Manitoba,
        R3A1B3,
        Canada,
        204-123-4567
        
            </span>
          </div>
        </label>
      </div>
    </div>
  </div>
  <div
    class="RadioButton RadioButton__AddressInfoContainer"
  >
    <div
      class="stx-field stx-field--is-disabled stx-field--with-radio-field RadioField"
    >
      <div
        class="stx-radio-field"
      >
        <span
          class="stx-radio stx-radio--is-disabled stx-radio-field__radio"
        >
          <input
            aria-checked="false"
            aria-disabled="true"
            class="stx-radio__input"
            disabled=""
            id="stxField11"
            name="shipping-address"
            role="radio"
            type="radio"
            value=""
          />
          <span
            class="stx-radio__box"
          />
        </span>
        <label
          class="stx-radio-field__label"
          for="stxField11"
        >
          <div
            class="CheckoutStep__FieldSetInfoContainer"
          >
            <span
              class="CheckoutStep__FieldSetInfo"
            >
              John Doe
            </span>
            <span
              class="CheckoutStep__FieldSetInfo CheckoutStep__FieldSetInfo--Secondary"
            >
              
        789 Fake Crescent,
        
        Winnipeg,
        Manitoba,
        R3A1B3,
        Canada,
        204-123-4567
        
            </span>
          </div>
        </label>
      </div>
    </div>
  </div>
  <div
    class="RadioButton RadioButton__NewAddressContainer"
  >
    <div
      class="stx-field stx-field--is-disabled stx-field--with-radio-field RadioField"
    >
      <div
        class="stx-radio-field"
      >
        <span
          class="stx-radio stx-radio--is-disabled stx-radio--is-checked stx-radio-field__radio"
        >
          <input
            aria-checked="true"
            aria-disabled="true"
            checked=""
            class="stx-radio__input"
            disabled=""
            id="stxField12"
            name="shipping-address"
            role="radio"
            type="radio"
            value=""
          />
          <span
            class="stx-radio__box"
          />
        </span>
        <label
          class="stx-radio-field__label"
          for="stxField12"
        >
          Add a new address
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ShippingAddress renders ShippingAddress component with selected address 1`] = `
<DocumentFragment>
  <div
    class="RadioButton RadioButton__AddressInfoContainer"
  >
    <div
      class="stx-field stx-field--with-radio-field RadioField"
    >
      <div
        class="stx-radio-field"
      >
        <span
          class="stx-radio stx-radio-field__radio"
        >
          <input
            aria-checked="false"
            class="stx-radio__input"
            id="stxField5"
            name="shipping-address"
            role="radio"
            type="radio"
            value=""
          />
          <span
            class="stx-radio__box"
          />
        </span>
        <label
          class="stx-radio-field__label"
          for="stxField5"
        >
          <div
            class="CheckoutStep__FieldSetInfoContainer"
          >
            <span
              class="CheckoutStep__FieldSetInfo"
            >
              John Doe
            </span>
            <span
              class="CheckoutStep__FieldSetInfo CheckoutStep__FieldSetInfo--Secondary"
            >
              
        123 Fake Street,
        Unit 12,
        Winnipeg,
        Manitoba,
        R3A1B3,
        Canada,
        204-123-4567
        
            </span>
          </div>
        </label>
      </div>
    </div>
  </div>
  <div
    class="RadioButton RadioButton__AddressInfoContainer"
  >
    <div
      class="stx-field stx-field--with-radio-field RadioField"
    >
      <div
        class="stx-radio-field"
      >
        <span
          class="stx-radio stx-radio--is-checked stx-radio-field__radio"
        >
          <input
            aria-checked="true"
            checked=""
            class="stx-radio__input"
            id="stxField6"
            name="shipping-address"
            role="radio"
            type="radio"
            value=""
          />
          <span
            class="stx-radio__box"
          />
        </span>
        <label
          class="stx-radio-field__label"
          for="stxField6"
        >
          <div
            class="CheckoutStep__FieldSetInfoContainer"
          >
            <span
              class="CheckoutStep__FieldSetInfo"
            >
              John Doe
            </span>
            <span
              class="CheckoutStep__FieldSetInfo CheckoutStep__FieldSetInfo--Secondary"
            >
              
        456 Test Drive,
        
        Winnipeg,
        Manitoba,
        R3A1B3,
        Canada,
        204-123-4567
        
            </span>
          </div>
        </label>
      </div>
    </div>
  </div>
  <div
    class="RadioButton RadioButton__AddressInfoContainer"
  >
    <div
      class="stx-field stx-field--with-radio-field RadioField"
    >
      <div
        class="stx-radio-field"
      >
        <span
          class="stx-radio stx-radio-field__radio"
        >
          <input
            aria-checked="false"
            class="stx-radio__input"
            id="stxField7"
            name="shipping-address"
            role="radio"
            type="radio"
            value=""
          />
          <span
            class="stx-radio__box"
          />
        </span>
        <label
          class="stx-radio-field__label"
          for="stxField7"
        >
          <div
            class="CheckoutStep__FieldSetInfoContainer"
          >
            <span
              class="CheckoutStep__FieldSetInfo"
            >
              John Doe
            </span>
            <span
              class="CheckoutStep__FieldSetInfo CheckoutStep__FieldSetInfo--Secondary"
            >
              
        789 Fake Crescent,
        
        Winnipeg,
        Manitoba,
        R3A1B3,
        Canada,
        204-123-4567
        
            </span>
          </div>
        </label>
      </div>
    </div>
  </div>
  <div
    class="RadioButton RadioButton__NewAddressContainer"
  >
    <div
      class="stx-field stx-field--with-radio-field RadioField"
    >
      <div
        class="stx-radio-field"
      >
        <span
          class="stx-radio stx-radio-field__radio"
        >
          <input
            aria-checked="false"
            class="stx-radio__input"
            id="stxField8"
            name="shipping-address"
            role="radio"
            type="radio"
            value=""
          />
          <span
            class="stx-radio__box"
          />
        </span>
        <label
          class="stx-radio-field__label"
          for="stxField8"
        >
          Add a new address
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
